pkg/katautils/config.go

```diff
@@ -595,6 +596,13 @@ func newQemuHypervisorConfig(h hypervisor) (vc.HypervisorConfig, error) {
 	kernelParams := h.kernelParams()
 	machineType := h.machineType()
 
+	// The "microvm" machine type doesn't support NVDIMM so override the
+	// config setting to explicitly disable it (i.e. don't require the
+	// user to add 'disable_image_nvdimm = true' in the .toml file).
+	if machineType == govmmQemu.MachineTypeMicrovm {
+		h.DisableImageNvdimm = true
```

Comment:
I think this is fine, but could you add a log?

Reply:
Sure. I'll make the changes below (adding a check for machineType so that a message is only printed when it's overridden).

```
--- a/pkg/katautils/config.go
+++ b/pkg/katautils/config.go
@@ -599,8 +599,10 @@ func newQemuHypervisorConfig(h hypervisor) (vc.HypervisorConfig, error) {
        // The "microvm" machine type doesn't support NVDIMM so override the
        // config setting to explicitly disable it (i.e. don't require the
        // user to add 'disable_image_nvdimm = true' in the .toml file).
-       if machineType == govmmQemu.MachineTypeMicrovm {
+       if machineType == govmmQemu.MachineTypeMicrovm &&
+               h.DisableImageNvdimm == false {
                h.DisableImageNvdimm = true
+               kataUtilsLogger.Info("Setting 'disable_image_nvdimm = true' as microvm does not support NVDIMM")
        }
 
        blockDriver, err := h.blockDeviceDriver()

```